test/src/tests-eCommerce.js

```diff
@@ -218,35 +219,34 @@ describe('eCommerce', function() {
             );
 
         mParticle.eCommerce.logPurchase(transactionAttributes, product);
-        var data = getEvent(mockServer.requests, 'eCommerce - Purchase');
+        var purchaseEvent1 = findEventFromRequest(window.fetchMock._calls, 'purchase');
+
+        window.fetchMock._calls = [];
 
-        mockServer.requests = [];
         mParticle.eCommerce.logProductAction(mParticle.ProductActionType.Purchase, product, null, null, transactionAttributes)
-        var data2 = getEvent(mockServer.requests, 'eCommerce - Purchase');
-
-        data.should.have.property('pd');
-
-        data.pd.an.should.equal(data2.pd.an);
-        data.pd.ti.should.equal(data2.pd.ti);
-        data.pd.ta.should.equal(data2.pd.ta);
-        data.pd.tcc.should.equal(data2.pd.tcc);
-        data.pd.tr.should.equal(data2.pd.tr);
-        data.pd.ts.should.equal(data2.pd.ts);
-        data.pd.tt.should.equal(data2.pd.tt);
-        data.pd.pl.length.should.equal(data2.pd.pl.length)
-
-        data.pd.pl[0].id.should.equal(data2.pd.pl[0].id)
-        data.pd.pl[0].pr.should.equal(data2.pd.pl[0].pr)
-        data.pd.pl[0].qt.should.equal(data2.pd.pl[0].qt)
-        data.pd.pl[0].br.should.equal(data2.pd.pl[0].br)
-        data.pd.pl[0].va.should.equal(data2.pd.pl[0].va)
-        data.pd.pl[0].ca.should.equal(data2.pd.pl[0].ca)
-        data.pd.pl[0].ps.should.equal(data2.pd.pl[0].ps)
-        data.pd.pl[0].cc.should.equal(data2.pd.pl[0].cc)
-        data.pd.pl[0].tpa.should.equal(data2.pd.pl[0].tpa)
-        data.pd.pl[0].nm.should.equal(data2.pd.pl[0].nm)
+        var purchaseEvent2 = findEventFromRequest(window.fetchMock._calls, 'purchase');
+
+        purchaseEvent1.data.product_action.action.should.equal(purchaseEvent2.data.product_action.action);
```

Comment:
Nitpick. Would it make sense to destructure both actions and compare those?

For example:
```
const { product_action: productAction1 } = purchaseEvent1.data;
const { product_action: productAction2 } = purchaseEvent2.data;

productAction1.transaction_id.should.equal(productAction2.transaction_id);
```

As an alternative, I think if you use `expect` instead of `should`, it might have matchers that can check all properties.

Reply:
good call.